File: src/task1/cpp_pubsub/src/sub_node.cpp
#include <memory>
#include <functional>
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"
using std::placeholders::_1;

class Subscriber_Node : public rclcpp::Node
{
    public:
        Subscriber_Node() : Node("subscriber_node")
        {
            subscription_=this->create_subscription<std_msgs::msg::String>("Channel2", 10, std::bind(&Subscriber_Node::listen_callback, this, _1));
        }
    private:
        void listen_callback(const std_msgs::msg::String &msg) const
        {
            RCLCPP_INFO(this->get_logger(), "Seconds Elapsed: '%s'", msg.data.c_str());
        }
        rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription_;
};

int main(int argc, char *argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<Subscriber_Node>());
    rclcpp::shutdown();
    return 0;
}

File: src/task1/py_pubandsub/py_pubandsub/sub_node.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class Subscriber_Node(Node):
    def __init__(self):
        super().__init__('Subscriber_Node')
        self.subscription = self.create_subscription(String, 'Channel1', self.listen_callback, 10)
        self.subscription

    def listen_callback(self, msg):
        self.get_logger().info(f'Seconds Elapsed:{msg.data}')

def main(args=None):
    rclpy.init(args=args)
    sub_node=Subscriber_Node()
    rclpy.spin(sub_node)
    sub_node.destory_node()
    rclpy.shutdown()

if __name__=='__main__':
    main()